File: Laba2/24.swift
print("Введите количество чисел", terminator: " ")
if let input = readLine(), let n = Int(input) {
    if n > 0 {
        for i in 0..<n {
            if let numInput = readLine(), var num = Int(numInput) {
                num = abs(num)
                var summ = 0
                var mult = 1
                // Обрабатываем 0 отдельно
                if num == 0 {
                    summ = 0
                    mult = 0
                } else {
                    var tempNum = num
                    while tempNum > 0 {
                        let digit = tempNum % 10
                        summ += digit
                        mult *= digit
                        tempNum = tempNum / 10
                    }
                }
                // Сравниваем сумму и произведение
                if summ < mult {
                    print(i)
                }
            }
        }
    } else {
        print("Некорректное количество чисел")
    }
} else {
    print("Некорректное количество чисел")
}

File: Laba2/24.py
n = int(input("Введите количество чисел "))
if n > 0:
    for i in range(n):
        num = abs(int(input()))
        summ = 0
        mult = 1
        # Обрабатываем 0 отдельно
        if num == 0:
            summ = 0
            mult = 0
        else:
            while num > 0:
                digit = num % 10
                summ += digit
                mult *= digit
                num = num // 10
        # Сравниваем сумму и произведение
        if summ < mult:
            print(i, end="\n")
else:
    print("Некорректное количество чисел")
